src/engine/strings.c

```diff
@@ -219,5 +225,25 @@ void string_unit_test()
         assert( copy->size == strlen( original ) );
         string_free( copy );
     }
+
+    {
+        char * const foo = "Foo    ";
+        string foo_copy[ 1 ];
+        string_copy( foo_copy, foo );
+        string_rtrim( foo_copy );
+        assert( !strcmp( foo_copy->value, "Foo" ) );
+
+        string_rtrim( foo_copy );
+        assert( !strcmp( foo_copy->value, "Foo" ) );
+
+        char * const bar = "Bar\0\0\0";
+        string bar_copy[ 1 ];
+        string_copy( bar_copy, bar );
+        string_rtrim( bar_copy );
+        assert( !strcmp( bar_copy->value, "Bar" ) );
+
+        string_rtrim( bar_copy );
+        assert( !strcmp( bar_copy->value, "Bar" ) );
+    }
```

Comment:
I do not know how to run these tests. I'm reasonably confident they work, but I do not know how to verify this. **How can I verify that these tests function as expected?**